src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.OneNote/Main.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using ManagedCommon;
+using Microsoft.PowerToys.Run.Plugin.OneNote.Properties;
+using ScipBe.Common.Office.OneNote;
+using Windows.Win32;
+using Windows.Win32.Foundation;
+using Windows.Win32.UI.WindowsAndMessaging;
+using Wox.Plugin;
+
+namespace Microsoft.PowerToys.Run.Plugin.OneNote
+{
+    /// <summary>
+    /// A power launcher plugin to search across time zones.
+    /// </summary>
+    public class Main : IPlugin, IPluginI18n
+    {
+        /// <summary>
+        /// A value indicating if the OneNote interop library was able to successfully initialize.
+        /// </summary>
+        private readonly bool _oneNoteInstalled;
+
+        /// <summary>
+        /// The initial context for this plugin (contains API and meta-data)
+        /// </summary>
+        private PluginInitContext? _context;
+
+        /// <summary>
+        /// The path to the icon for each result
+        /// </summary>
+        private string _iconPath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Main"/> class.
+        /// </summary>
+        public Main()
+        {
+            UpdateIconPath(Theme.Light);
+            try
+            {
+                _ = OneNoteProvider.PageItems.Any();
+                _oneNoteInstalled = true;
+            }
+            catch (COMException)
+            {
+                // OneNote isn't installed, plugin won't do anything.
+                _oneNoteInstalled = false;
+            }
```

Comment:
Should we log a warning here? Should this be moved to the init method?

Reply:
I'm not sure if this should have a warning, it's expected behavior when the user doesn't have OneNote installed, I'll let the repo devs decide if that's warning level or not. @stefansjfw?

Reply:
I have no objection to moving it to the Init method, will do

Reply:
I am unsure if it is called more times than needed if it's in main. Thoughts, @stefansjfw?

Reply:
this plugin is off by default, so I'd guess user will enable it only if OneNote is installed .. So, this should be fine without warning. 

> I am unsure if it is called more times than needed if it's in main. Thoughts, @stefansjfw?

Main should be called only on plugin creation, right?

Reply:
Yes it's only called once, in PluginPair.cs:171